### PR TITLE
limit regression test result search to jwst files

### PR DESCRIPTION
**Pull request by braingram**
This PR updates the jenkins configuration files to limit what files are considered for uploading during the `Upload artifacts` step. The changes here are the same as those applied in this romancal PR:
https://github.com/spacetelescope/romancal/pull/950

If/when jwst adds `astroquery` (or `webbpsf` which depends on `astroquery`) the jenkins runs will fail on the `Upload artifacts` step due to searching files beyond those that are generated when attempting to upload test results to artifactory. This search will fail due to finding the [mission_results.json](https://github.com/astropy/astroquery/blob/5add2e3b0ff5b56630840a0f57196c19054f5c01/astroquery/mast/tests/data/mission_results.json#L5) file in the astroquery package.

This PR limits the files searched to those generated during the pytest run in the jenkins workflow.

Regression tests passed with no errors: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1015/

**Checklist for maintainers**
- [ ] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [ ] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
Is this ready to merge?

**Comment by braingram** [comment]
Thanks for taking a look and following up. This is ready to merge. Should we wait to rebase after #8032 is merged?

**Comment by hbushouse** [comment]
> Thanks for taking a look and following up. This is ready to merge. Should we wait to rebase after #8032 is merged?

Yes, let's wait for that other PR to get merged, then rebase and merge this one.

**Pull request merged**
